### PR TITLE
Overflow hittest

### PR DESCRIPTION
**Pull request by osdnk**
Properly deliver touches to views that extend beyond parent's bounds with visible overflow.

This PR relies on changes made to RN core and available since 0.57.2 https://github.com/facebook/react-native/commit/bc973d20e01abaf89bd792c667909f2180df645e

### CODE REVIEW

File: android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java [hunk 1]
```diff
@@ -384,7 +380,7 @@ private boolean recordViewHandlersForPointer(View view, float[] coords, int poin
     if (handlers != null) {
       for (int i = 0, size = handlers.size(); i < size; i++) {
         GestureHandler handler = handlers.get(i);
-        if (handler.isEnabled() && handler.isWithinBounds(view, coords[0], coords[1])) {
+        if (handler.isEnabled() && (handler.isWithinBounds(view, coords[0], coords[1]) || !isClipping(view))) {
```
[the remark this answers is not in the document]
Comment (osdnk) [reply android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java]: Not sure about it. Don't we want to handle case if some inner view which could be outside the parent can handle parent's gesture. If yes, I believe, it's necessary.

[the remark this answers is not in the document]
Comment (kmagiera) [reply android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java]: If you want handler to activate for some other child view you set it on that child view and not on a parent

[the remark this answers is not in the document]
Comment (osdnk) [reply android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java]: If you're fine with this approach, it's nod needed to add this confition

[the remark this answers is not in the document]
Comment (kmagiera) [reply android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandlerOrchestrator.java]: handler should activate only when user is interactive with a view it is attached to

File: android/src/main/java/com/swmansion/gesturehandler/react/RNViewConfigurationHelper.java [hunk 2]
```diff
@@ -45,4 +46,16 @@ public View getChildInDrawingOrderAtIndex(ViewGroup parent, int index) {
     }
     return parent.getChildAt(index);
   }
+
+  @Override
+  public boolean isViewClippingChildren(ViewGroup view) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2 && !view.getClipChildren()) {
+        if (view instanceof ReactViewGroup) {
+            String overflow =((ReactViewGroup) view).getOverflow();
```
Comment (kmagiera) [inline android/src/main/java/com/swmansion/gesturehandler/react/RNViewConfigurationHelper.java]:
```suggestion
            String overflow = ((ReactViewGroup) view).getOverflow();
```

**Pull request merged**